strongbox-web-core/src/main/java/org/carlspring/strongbox/controllers/security/ldap/LdapAuthenticatorConfigurationController.java

```diff
@@ -0,0 +1,352 @@
+                            @ApiResponse(code = 204, message = "Group search filter was not provided."),
+                            @ApiResponse(code = 400, message = "LDAP is not enabled or groupSearchFilter is not supported via this method.") })
+    @RequestMapping(value = "/groupSearchFilter", method = RequestMethod.GET, produces = { MediaType.APPLICATION_XML_VALUE })
+    public ResponseEntity<String> getGroupSearchFilter()
+            throws JAXBException
+    {
+        if (!springSecurityLdapInternalsSupplier.isLdapAuthenticationEnabled())
+        {
+            return toXmlError(LdapMessages.NOT_CONFIGURED, HttpStatus.BAD_REQUEST);
+        }
+        LdapAuthoritiesPopulator populator = springSecurityLdapInternalsSupplier.getAuthoritiesPopulator();
+        if (populator == null)
+        {
+            return ResponseEntity.noContent().build();
+        }
+        if (!(populator instanceof DefaultLdapAuthoritiesPopulator))
+        {
+            return ResponseEntity.badRequest().body(
+                    "Unable to display groupSearchFilter configuration. " + populator.getClass() +
+                    " is not supported via this method.");
+        }
+
+        final GenericParser<LdapGroupSearchXmlHolder> parser = new GenericParser<>(LdapGroupSearchXmlHolder.class);
+        final String result = parser.serialize(
+                springSecurityLdapInternalsSupplier.ldapGroupSearchHolder((DefaultLdapAuthoritiesPopulator) populator));
+
+        return ResponseEntity.ok(result);
+    }
+
+    @ApiOperation(value = "Updates LDAP group search filter. See http://docs.spring.io/spring-security/site/docs/current/reference/html/ldap.html#loading-authorities")
+    @ApiResponses(value = { @ApiResponse(code = 200, message = "Group search filter updated."),
+                            @ApiResponse(code = 204, message = "LdapAuthoritiesPopulator was not provided."),
+                            @ApiResponse(code = 400, message = "LDAP is not enabled or ldapAuthoritiesPopulator class is not supported via this method.") })
+    @RequestMapping(value = "/groupSearchFilter/{searchBase}/{searchFilter}", method = RequestMethod.PUT)
+    public ResponseEntity<String> updateGroupSearchFilter(@PathVariable String searchBase,
+                                                          @PathVariable String searchFilter)
+    {
+        if (!springSecurityLdapInternalsSupplier.isLdapAuthenticationEnabled())
+        {
+            return ResponseEntity.badRequest().body(LdapMessages.NOT_CONFIGURED);
+        }
+        LdapAuthoritiesPopulator ldapAuthoritiesPopulator = springSecurityLdapInternalsSupplier.getAuthoritiesPopulator();
+        if (ldapAuthoritiesPopulator == null)
+        {
+            return ResponseEntity.noContent().build();
+        }
+        if (!(ldapAuthoritiesPopulator instanceof DefaultLdapAuthoritiesPopulator))
+        {
+            return ResponseEntity.badRequest().body(
```

Comment:
Cosmetic: Please, see my previous comment (and apply it wherever else necessary). Thanks! :)